src/scanner/mod.rs

```diff
@@ -100,41 +103,59 @@ impl Scanner {
     ///
     /// Note: `self` must contain `self.ip`.
     async fn scan_socket(&self, socket: SocketAddr) -> io::Result<SocketAddr> {
-        match self.connect(socket).await {
-            Ok(x) => {
-                debug!(
-                    "Connection was successful, shutting down stream {}",
-                    &socket
-                );
-                match x.shutdown(Shutdown::Both) {
-                    Err(e) => debug!("Shutdown stream error {}", &e),
-                    _ => {}
-                }
-                if !self.greppable {
-                    if self.accessible {
-                        println!("Open {}", socket.to_string());
-                    } else {
-                        println!("Open {}", socket.to_string().purple());
+        let mut ret = Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("Did not connect to {}", &socket),
+        ));
+
+        for nr_try in 0..self.tries {
```

Comment:
Since you always use `nr_try`, this could be an iter over `1..=self.tries`

Reply:
I corrected it on my side.